lib/components/src/controls/helpers.test.ts

```diff
@@ -0,0 +1,22 @@
+import { getControlId } from './helpers';
+
+describe('getControlId', () => {
+  [
+    // testName, input, output
+    ['empty string', ' ', 'control--'],
+    ['lower case', 'some-id', 'control-some-id'],
+    ['upper case', 'SOME-ID', 'control-SOME-ID'],
+    ['handles basic substitution', 'a b$c?d😀e', 'control-a-b$c?d😀e'],
+    ['cyrillic', 'Кириллический идентификатор', 'control-Кириллический-идентификатор'],
+    ['korean', '바보', 'control-바보'],
+    [
+      'weird',
+      'some ,’–—―′¿`" weird <>()!.!!!{}[] id %^&$*#& 😀',
+      'control-some-,’–—―′¿`"-weird-<>()!.!!!{}[]-id-%^&$*#&-😀',
+    ],
+  ].forEach(([testName, input, output]) => {
```

Comment:
Thanks @YozhEzhi ! A couple things:

1) can you convert this to jest `it.each` which is a more standard/idiomatic way of doing this: https://jestjs.io/docs/api#testeachtablename-fn-timeout

2) a bunch of these are not valid DOM id's:

> ID and NAME tokens must begin with a letter ([A-Za-z]) and may be followed by any number of letters, digits ([0-9]), hyphens ("-"), underscores ("_"), colons (":"), and periods (".").

https://stackoverflow.com/questions/1077084/what-characters-are-allowed-in-dom-ids#:~:text=The%20W3C%20spec%20Basic%20HTML,%22).%22

Reply:
@shilman 
1. I'll do.
2. Is your advise to remove non valid cases? Ok. I'll do.

Reply:
Done.

Reply:
No, my point is that users can name their controls anything including non-valid characters. But the conversion function needs to turn those into IDs that only contain valid characters. So the conversion function needs to be a little smart to do that properly.

Reply:
🤔

Reply:
This is not as simple as I thought :)

For example: What if user named control "Hello 😄" and "Hello 😄". Id should be unique. Now I have no idea how to deal that situation.

Reply:
> What if user named control "Hello 😄" and "Hello 😄"

Then the users is going to discover that one of those controls does not show up?, If we use this Id as a react key, react will log a warning and render only 1 of them.

Reply:
Guys, as I see in `next`[ branch code](https://github.com/storybookjs/storybook/blob/next/lib/components/src/controls/options/Checkbox.tsx#L69): some id's are generated like this:
```
const id = `${name}-${key}`;
```

And id becomes like `arg-Some value`. It's not valid.

In this PR I've made id generation to be like:
```
const id = `${name}-${index}`;
```

Example result:

<img width="577" alt="2" src="https://user-images.githubusercontent.com/2706062/116793606-528fb000-aad0-11eb-9a69-afc2e646c816.png">

Reply:
@shilman @ndelangen
**IMHO**, my solution **does not degrade** current behaviour of controls. And it brings **some** improvements.
But maybe, I don't understand some usecases of controls.

So, please, feel free to close this PR if my solution will not work as it should to be.